velox/exec/tests/DriverTest.cpp

```diff
@@ -235,24 +235,31 @@ class DriverTest : public OperatorTestBase {
     std::lock_guard<std::mutex> l(wakeupMutex_);
     if (!wakeupInitialized_) {
       wakeupInitialized_ = true;
-      wakeupThread_ = std::thread([&]() {
+      wakeupThread_ = std::thread([this]() {
         int32_t counter = 0;
         for (;;) {
-          if (wakeupCancelled_) {
-            return;
+          {
+            std::lock_guard<std::mutex> l(wakeupMutex_);
+            if (wakeupCancelled_) {
+              return;
+            }
           }
           // Wait a small interval and realize a small number of queued
           // promises, if any.
           auto units = 1 + (++counter % 5);
+
           // NOLINT
           std::this_thread::sleep_for(std::chrono::milliseconds(units));
-          auto count = 1 + (++counter % 4);
-          for (auto i = 0; i < count; ++i) {
-            if (wakeupPromises_.empty()) {
-              break;
+          {
+            std::lock_guard<std::mutex> l(wakeupMutex_);
```

Comment:
@oerling This shadows variable on line 235 and will also deadlock because of it.